packages/opentelemetry-basic-tracer/src/BasicTracer.ts

```diff
@@ -54,42 +54,19 @@ export class BasicTracer implements types.Tracer {
    * decision.
    */
   startSpan(name: string, options: types.SpanOptions = {}): types.Span {
-    let parentSpanContext: types.SpanContext | undefined;
-
-    // parent is a SpanContext
-    if (options.parent && (options.parent as types.SpanContext).traceId) {
-      parentSpanContext = options.parent as types.SpanContext;
-    }
-    // parent is a Span
-    if (
-      options.parent &&
-      typeof (options.parent as types.Span).context === 'function'
-    ) {
-      parentSpanContext = (options.parent as types.Span).context();
-    }
-
+    const parentSpanContext = this._getParentSpanContext(options.parent);
     // make sampling decision
     if (!this._sampler.shouldSample(parentSpanContext)) {
       // TODO: propagate SpanContext, for more information see
       // https://github.com/open-telemetry/opentelemetry-js/pull/99#issuecomment-513325536
       return NOOP_SPAN;
     }
 
-    // span context
-    const traceId = parentSpanContext
-      ? parentSpanContext.traceId
-      : randomTraceId();
-    const spanId = randomSpanId();
-
-    // TODO: create a real Span
-    const span = new NoopSpan({
-      traceId,
-      spanId,
-    });
+    options.parent = parentSpanContext;
```

Comment:
Looking at this in the context of the tracer, I'm curious what happens when the parentSpanContext not provided in options and this is null. How does sampling occur if the span is responsible for creating the span context? I'm now thinking that generating the trace should be the responsibility of the tracer. LightStep tracer implementations allow a context to passed with just a trace id, letting the span know it's a root span and should generate its own ID.

Reply:
Even I think generating the `traceId` and `spanId` should be the  responsibility of the tracer and we should pass these information to sampling function to make the sampling decision (_[these](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-sampling.md#shouldsample) are the required aguments for making sampling decision as per specs_). Also, along with `SpanOptions` we should pass these ids while creating the `Span` instance.

Reply:
Shouldn't we create a new option here for the constructor here in an immutable way? Otherwise, we override the parent span. Like `const spanOptions = Object.assign({}, options, { parent: parentSpanContext })`

Reply:
Hmm, good point. Thanks for the suggestion, I made the required change. ptal and approve if all looks good.